xdebug_stack.c

```diff
@@ -1288,6 +1312,7 @@ function_stack_entry *xdebug_add_stack_frame(zend_execute_data *zdata, zend_op_a
 					} else {
 						ZVAL_COPY_VALUE(&(tmp->var[tmp->varc].data), ZEND_CALL_VAR_NUM(zdata, zdata->func->op_array.last_var + zdata->func->op_array.T + i - arguments_wanted));
 					}
+					gc_locker(tmp, &(tmp->var[tmp->varc].data));
```

Comment:
It would make more sense to me if the above ZVAL_COPY_VALUES were ZVAL_COPY and had corresponding ptr_dtors. The whole object GC thing sounds like a red herring.

Reply:
I'm not good enough in PHP code code to get that done well, I don't know what "ptr_dtors" would refer to.

As for the "why only objects?" thing, I couldn't make a failing test for arrays and such https://github.com/xdebug/xdebug/pull/474#discussion_r284823304 So, I too think the arrays could fail too, but I haven't met these case so far.

Reply:
My first guess is that you were testing with immutable arrays or interned strings. The values need to look "sufficiently dynamic", such as `$foo = "foo"; $foo .= "bar"` or `$foo = [1]; $foo[] = 2`.

By ptr_dtors I meant using the zval_ptr_dtor() function.

Reply:
Having looked at your existing tests you'd also want it to be a temporary, so something like `$foo = "foo"; call($foo . "bar")` and `$foo = [1]; call($foo + [2])` might do it.